server/routerlicious/packages/services-core/src/checkpointService.ts

```diff
@@ -31,62 +31,102 @@ export class CheckpointService implements ICheckpointService {
 		service: DocumentLambda,
 		checkpoint: IScribe | IDeliState,
 		isLocal: boolean = false,
+		markAsCorrupt: boolean = false,
+	) {
+		// services may not be able to process documents in corrupted state
+		// so we write to local and global databases when marking document as corrupt
+		if (markAsCorrupt) {
+			await this.writeGlobalCheckpoint(
+				documentId,
+				tenantId,
+				service,
+				checkpoint,
```

Comment:
Just to confirm: do we also include corrupted state information in the `checkpoint` payload here?

Reply:
Correct